### PR TITLE
Issue 2466,2467,2468 - document secret and secretBinding for service,…

### PR DESCRIPTION
**Pull request by linggao**
… pattern and deployment policy

Signed-off-by: linggao <linggao@us.ibm.com>

### CODE REVIEW

File: cli/exchange/business.go [hunk 1]
```diff
@@ -281,6 +281,20 @@ func BusinessNewPolicy() {
...
+		`    }`,
+		`  ],`,
+		`  "secretBinding": [ /* ` + msgPrinter.Sprintf("A list of secret bindings for the secret names defined in the services, listed by service.") + ` */`,
+		`    {           `,
+		`      "serviceOrgid": "",         /* ` + msgPrinter.Sprintf("The org of the service.") + ` */`, 
+		`      "serviceUrl": "",           /* ` + msgPrinter.Sprintf("The name of the service.") + ` */`,
+		`      "serviceVersionRange": "",  /* ` + msgPrinter.Sprintf("The service version range.") + ` */`,
+		`      "secrets": [                /* ` + msgPrinter.Sprintf("The secret bindings.") + ` */`,
+		`        {      `,
+		`          "<service-secret-name>": "<vault-secret-name>"  /* ` + msgPrinter.Sprintf("The valid formats for vault secret names are:") + ` */`,
```
Comment (dabooz) [inline cli/exchange/business.go:295]: Please change vault-secret-name to secret-provider-secret-name. And change the Sprintf string too.

File: docs/deployment_policy.md [hunk 2]
```diff
@@ -34,12 +34,19 @@ Following are the fields in the JSON representation of a deployment policy:
   - `inputs`: A list of service variables to set.
     - `name`: The name of the variable. This is the same as a variable name found in `userInputs` as defined [here](./service_def.md).
     - `value`: The value to be assigned to the variable. Service variables are typed as described in `userInputs` defined [here](./service_def.md).
+- `secretBinding`: This section is used to bind secret names defined in the servcie with the secret names in the vault. The secret value will be retrived from the vault and passed to the service container at the deployment time. The secret value is used by the service container to access other applications.
```
Comment (dabooz) [inline docs/deployment_policy.md:37]: typo servcie
replace references to vault with secret provider

File: docs/deployment_policy.md [hunk 3]
```diff
@@ -34,12 +34,19 @@ Following are the fields in the JSON representation of a deployment policy:
...
+  - `serviceUrl`: The name of the service. It can be the top level services defined in the `services` attribute or their dependent services. This is the same value as found in the `url` field [here](./service_def.md).
```
Comment (dabooz) [inline docs/deployment_policy.md:38]: 'or their dependent services.' change to 'or one of its dependency services.'

File: docs/deployment_policy.md [hunk 4]
```diff
@@ -34,12 +34,19 @@ Following are the fields in the JSON representation of a deployment policy:
...
+  - `serviceOrgid`: The organization in which the service in `serviceUrl` is defined.
+  - `serviceArch`: The hardware architecture of the service in `serviceUrl`, or `*` to indicate any compatible architecture. This is the same value as found in the `arch` field [here](./service_def.md).
+  - `serviceVersionRange`: A version range indicating the set of service versions to which this secret binding should be applied.
+  - `secrets`: A list of secret bindings. Each elelment is a map of string keyed by the name of the secret in the service. The value is the name of the secret in the vault. The valid formats for the vault secret names are: `<secretname>` for the organization level secret; `user/<username>/<secretname>` for the user level secret. The fully qualified vault secret name will be the the value defined here preceded by `openhorizon/<orgname>/` where the `<orgname>` is the organzation name of this deployment policy. However, it is invalid to specify the fully qualified name here.
```
Comment (dabooz) [inline docs/deployment_policy.md:42]: more vault references. 
Also, please delete the sentence that starts "The fully qualified vault secret name will be the the value defined here preceded by `openhorizon/<orgname>/` where the `<orgname>` is the organzation name of this deployment policy. However, it is invalid to specify the fully qualified name here."

File: docs/deployment_policy.md [hunk 5]
```diff
@@ -34,12 +34,19 @@ Following are the fields in the JSON representation of a deployment policy:
...
 
 The following is an example of a deployment policy that deploys a service called `my.company.com.service.this-service`.
 The service is defined within organization `yourOrg`.
 This policy will deploy the service to any node which matches one of the architectures for which the service is defined, and is also compatible with nodes that have the property `aNodeProperty` set to `someValue`.
 Two versions of the service are mentioned, with version `2.3.1` having a higher priority for dpeloyment than version `2.3.0`.
 The deployed service is dependent on service `my.company.com.service.other` which has a variable `var1` that needs to be set in order for it to deploy correctly.
+Both `2.3.0` and `2.3.1` versions of the services have a secret `ai_secret` defined that the service container will use to access an AI service on the clound once the vault secret name is bound to it. The policy binds it to a vault secret named `cloud_ai_secret_name`.
```
Comment (dabooz) [inline docs/deployment_policy.md:49]: typo clound
and more vault references

File: docs/deployment_string.md [hunk 6]
```diff
@@ -39,6 +39,7 @@ Because Horizon uses the Docker API to start the containers on an edge node, man
     - `max_memory_mb`: `4096` - the maximum amount of memory the service's container can use
     - `max_cpus`: `1.5` - how much of the available CPU resources ther service's container can use. For instance, if the host machine has two CPUs and you set value to 1.5, the container is guaranteed to use at most one and a half of the CPUs
     - `log_driver`: the logging driver (e.g. `json-file`) to use for container logs, instead of default one (syslog)
+    - `secrets`: `{"ai_secret": {"description": "The token for cloud AI service."}, "sql_secret": {}}` - a list of secret names and the descriptions. The `description` can be omitted. A secret name is just a user defined string. A pattern or a deployment policy will associate it with the name of the secret in the vault. The horizon agent will attaches a secret from the vault to the deployment of the service.
```
Comment (dabooz) [inline docs/deployment_string.md:42]: Please change the last sentence to:
"The horizon agent will mount the secrets at '/open-horizon-secrets' within the service's containers. Each secret name appears as a file in that directory, containing the details of the secret from the secret provider. Each secret file is a JSON encoded file containing the "key" and "value" set when the secret was created with the hzn secretsmanager secret add command.

**Review by dabooz — CHANGES_REQUESTED** [review]

File: cli/exchange/business.go (hunk 1 above)
Comment (linggao) [reply cli/exchange/business.go:295]: done

File: docs/deployment_policy.md (hunk 2 above)
Comment (linggao) [reply docs/deployment_policy.md:37]: done

File: docs/deployment_policy.md (hunk 3 above)
Comment (linggao) [reply docs/deployment_policy.md:38]: done

File: docs/deployment_policy.md (hunk 4 above)
Comment (linggao) [reply docs/deployment_policy.md:42]: done

File: docs/deployment_policy.md (hunk 5 above)
Comment (linggao) [reply docs/deployment_policy.md:49]: done

File: docs/deployment_string.md (hunk 6 above)
Comment (linggao) [reply docs/deployment_string.md:42]: done

**Review by dabooz — APPROVED** [review]

**Pull request merged**